### PR TITLE
Improve audit log entry collection

### LINKED ISSUE #46: Audit log entries not appearing
If you look at the `audit_log` and `audit_log_changes` table, you see that the entries listed don't seem to line up with what the client shows. Checking the `audit_log_crawl` table shows that the last entry crawled was ID `365824096744767488`, which corresponds to `2017-10-06 11:34:21.347000 UTC` which seems a bit out of date.

The audit log and channel history crawlers are both using the same logic, so something weird is happening here.

Some ideas:
* something's wrong with the audit log part of the crawling
* audit log history fetching has some sort of fundamental difference in how it behaves
* the scheduling of the coroutines is preventing the audit log crawler from ever doing any work

### PR DESCRIPTION
**Pull request by ammongit**
Fixes #46 

An invalid check was making the crawler think the source was prematurely exhausted. Compare the new code with how the history crawler checks IDs.

### CODE REVIEW

**Comment by raylu** [comment]
how did the bug happen? is the last id we have sometimes greater than the last id of the new entries that we do want to ingest?

**Comment by ammongit** [comment]
@raylu It's because the results from `guild.audit_logs()` are unusual in that they are different from `channel.history()`. Instead of just giving the entries in that time range, it will sort of ignore your parameters if that time range has no entries in it, and instead give you the latest batch as if you had requested the most recent ones.

The original fix for this was to compare the `last_id` stored versus the last (biggest) ID in the entries it gave you. If you didn't do this check, then you get caught in an infinite loop, endlessly pulling the same batch of entries. However the check was flawed, meaning that we _never_ considered entries, even when they are new.

**Comment by ammongit** [comment]
Seeing as the JSON type postgres has allowed us to massively shrink the `audit_log_changes` table, it was basically not needed. I have merged in the "before" and "after" attributes into the main `audit_log` table. This also has improvements to how entries are stored, such as differentiating between null values and nonexistent ones.

File: statbot/crawler.py [hunk 1]
```diff
@@ -252,12 +252,12 @@ def __init__(self, client, sql, config, logger=null_logger):
         self.logger.info(f"Starting from ID {last_id} ({last})")
 
         entries = await guild.audit_logs(after=last, limit=limit).flatten()
```
Comment (raylu) [inline statbot/crawler.py]: can you leave a comment about how `after=last` doesn't exclusively give you audit log entries after the one you request and thus the silly `!=` behavior?

**Pull request merged**